xmodule/library_root_xblock.py

```diff
@@ -55,13 +55,20 @@ def author_view(self, context):
         Renders the Studio preview view.
         """
         fragment = Fragment()
-        self.render_children(context, fragment, can_reorder=False, can_add=True)
+        self.render_children(
```

Comment:
When the user lacks edit permission, the legacy library render page acts kinda derpy. It has buttons for things you're not allowed to do, and clicking those buttons shows you Studio errors. This change just forwards some of those permissions to the frontend.

I originally made these changes in service of the "legacy libraries are read-only after being migrated" idea.   I ended up dropping that from the scope of this PR, but these changes are good anyway.

I can remove these changes from the diff if any reviewer would prefer that.